pkg/controllers/disruption/singlenodeconsolidation_test.go

```diff
@@ -0,0 +1,303 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disruption_test
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/controllers/disruption"
+	"sigs.k8s.io/karpenter/pkg/test"
+	. "sigs.k8s.io/karpenter/pkg/test/expectations"
+	"sigs.k8s.io/karpenter/pkg/utils/pdb"
+)
+
+var nodePool1, nodePool2, nodePool3 *v1.NodePool
+var consolidation *disruption.SingleNodeConsolidation
+var nodePoolMap map[string]*v1.NodePool
+var nodePoolInstanceTypeMap map[string]map[string]*cloudprovider.InstanceType
+
+var _ = Describe("SingleNodeConsolidation", func() {
+	BeforeEach(func() {
+		nodePool1 = test.NodePool(v1.NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nodepool-1",
+			},
+			Spec: v1.NodePoolSpec{
+				Disruption: v1.Disruption{
+					ConsolidationPolicy: v1.ConsolidationPolicyWhenEmptyOrUnderutilized,
+					ConsolidateAfter:    v1.MustParseNillableDuration("0s"),
+				},
+			},
+		})
+		nodePool2 = test.NodePool(v1.NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nodepool-2",
+			},
+			Spec: v1.NodePoolSpec{
+				Disruption: v1.Disruption{
+					ConsolidationPolicy: v1.ConsolidationPolicyWhenEmptyOrUnderutilized,
+					ConsolidateAfter:    v1.MustParseNillableDuration("0s"),
+				},
+			},
+		})
+		nodePool3 = test.NodePool(v1.NodePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "nodepool-3",
+			},
+			Spec: v1.NodePoolSpec{
+				Disruption: v1.Disruption{
+					ConsolidationPolicy: v1.ConsolidationPolicyWhenEmptyOrUnderutilized,
+					ConsolidateAfter:    v1.MustParseNillableDuration("0s"),
+				},
+			},
+		})
+		ExpectApplied(ctx, env.Client, nodePool1, nodePool2, nodePool3)
+
+		// Set up NodePool maps for candidate creation
+		nodePoolMap = map[string]*v1.NodePool{
+			nodePool1.Name: nodePool1,
+			nodePool2.Name: nodePool2,
+			nodePool3.Name: nodePool3,
+		}
+		nodePoolInstanceTypeMap = map[string]map[string]*cloudprovider.InstanceType{
+			nodePool1.Name: {leastExpensiveInstance.Name: leastExpensiveInstance},
+			nodePool2.Name: {leastExpensiveInstance.Name: leastExpensiveInstance},
+			nodePool3.Name: {leastExpensiveInstance.Name: leastExpensiveInstance},
+		}
+
+		// Create a single node consolidation controller
+		baseConsolidation := disruption.MakeConsolidation(fakeClock, cluster, env.Client, prov, cloudProvider, recorder, queue)
```

Comment:
nit: I don't see you using this anywhere other than the line below -- do you need to assign this to a variable?